src/test/java/org/mockitousage/verification/DelayedExecution.java

```diff
@@ -1,54 +1,78 @@
+    /**
+     * Defines the number of milliseconds we expecting a Thread might need to unpark, we use this to avoid "oversleeping" while awaiting the deadline for  
+     */
+    private static final long MAX_EXPECTED_OVERSLEEP_MILLIS = 50;
     private final ScheduledExecutorService executor;
-    private final IMethods mock;
-    private final int delay;
-    private final TimeUnit timeUnit;
-    private final ArrayList<Character> invocations = new ArrayList<Character>();
-
-    DelayedExecution(ScheduledExecutorService executor, IMethods mock, int delay, TimeUnit timeUnit) {
-        this.executor = executor;
-        this.mock = mock;
-        this.delay = delay;
-        this.timeUnit = timeUnit;
+
+    public DelayedExecution() {
+        this.executor = newScheduledThreadPool(CORE_POOL_SIZE, maxPrioThreadFactory());
     }
 
-    void recordAsyncCall(char c) {
-        invocations.add(c);
+    public void callAsync(long delay, TimeUnit timeUnit, Runnable r) {
+        long deadline = timeUnit.toMillis(delay) + currentTimeMillis();
+
+        executor.submit(delayedExcution(r, deadline));
     }
 
-    void allAsyncCallsStarted() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(invocations.size());
-        for (final Character invocation : invocations) {
-            executor.execute(runnable(countDownLatch, invocation));
+    public void close() throws InterruptedException {
+        executor.shutdownNow();
+
+        if (!executor.awaitTermination(5, SECONDS)) {
+            throw new IllegalStateException("This delayed excution did not terminated after 5 seconds");
         }
-        countDownLatch.await();
     }
 
-    private Runnable runnable(final CountDownLatch countDownLatch, final Character invocation) {
+    private static Runnable delayedExcution(final Runnable r, final long deadline) {
         return new Runnable() {
             @Override
             public void run() {
-                countDownLatch.countDown();
-                sleep();
-                mock.oneArg(invocation.charValue());
+                //we park the current Thread till 50ms before we want to execute the runnable 
+                parkUntil(deadline - MAX_EXPECTED_OVERSLEEP_MILLIS);
+                //now we closing to the deadline by burning CPU-time in a loop
+                burnRemaining(deadline);
+
+                System.out.println(currentTimeMillis() - deadline);
+
+                r.run();
+            }
+
+            /**
+             * Loop in tight cycles until we reach the dead line. We do this cause sleep or park is very not precise, 
+             * this can causes a Thread to under- or oversleep, sometimes by +50ms.  
+             */
+            private void burnRemaining(final long deadline) {
+                long remaining;
+                do {
+                    remaining = deadline - currentTimeMillis();
+                } while (remaining > 0);
```

Comment:
I experiment also with a `sleep(1)`, ` sleep(0,500)`,  `sleep(0,100)` and `sleep(0,100)`. The precision of the sleep implementation in the JVM or OS seems not to be very accurate. In average the sleep calls oversleeps by 8ms and up to ~50ms, which also depends on the Thread priority. 

Anyway, I think the real problem is that we don't have a realtime JVM. The timer resolution on Win is pretty low (15,6ms), GC's may occure and thread scheduling is not deterministic. All of this lead to flickering tests. Maybe it is better to throw an exception in case a value <30ms is passed to timeout().